### ISSUE TITLE
Any easier/faster way to obtain data?

### ISSUE THREAD

**Issue by piee-kun** [CakeCrusher/mimicbot#8]
I'm trying to use a dataset of 100k of my own messages among 3.5mil of other messages. The data collection takes over 5hr, but I'm not letting it finish due to time constraints. Is there any faster way to obtain training data as opposed to mining it from the server itself?

**Comment by CakeCrusher** [comment]
@piee-kun 
Off the top of my head I know of one alternative, which is basically a web scraper but I would imagine that is even slower (I'll link this solution later today).
Another thing you could do is request (download) your data from discord itself (assuming that they offer it, I'll look into that later today), preprocess it as necessary and reformat it so you train it with your own data. On the README's video you can see how that works.

**Comment by piee-kun** [comment]
I already have both my data and a DiscordChatExporter capture of the entire 3.5mil message channel. Is that able to be used or not?

**Comment by CakeCrusher** [comment]
@piee-kun 
Yes its able to be used but not out of the box you need to format the data to fulfill these requirements:
1. Create a `messages.csv` file with two columns `author_id` (id of user) and `content` (message of user)
2. The rows must be organized from newest to oldest so that the newest message is of index 0
3. Then simply run the `forge` command and set it as "individually ran bot" (which will require your messages file)
4. ![image](https://user-images.githubusercontent.com/37946988/227028033-b4eaa88c-e1ee-42bc-82dd-629fcab7d916.png)
5. Fill the rest of the data.
6. ![image](https://user-images.githubusercontent.com/37946988/227026492-6cdbcd39-a3dd-45f7-89f4-c282a3ca90ba.png)

The pipeline will then train the model with your `messages` file.

**Comment by piee-kun** [comment]
Thanks so much for the info and for the time out of your day in order to help me. I'll get back to you with the results.

**Comment by piee-kun** [comment]
Well, I've formatted the data properly and now I'm stuck on step 2 of initializing, I left it for basically the entire day and nothing came of it.

**Comment by CakeCrusher** [comment]
@piee-kun Im sorry to hear that. Could you please share the logs and anything else you think may be useful to figure out what is wrong?

At the current state Mimicbot is not too fault tolerant so I can imagine a couple things that could go wrong but your logs will help a lot.

**Comment by piee-kun** [comment]
Where are the logs stored? I see none generated.

**Comment by piee-kun** [comment]
Okay scratch that, I got to training and it's doing that.
![image](https://user-images.githubusercontent.com/31517004/228958395-4783c745-b2cd-4a24-aac5-eafb6a14b76a.png)

And when it does finish I get this error ``huggingface_hub.utils._validators.HFValidationError: Repo id must use alphanumeric chars or '-', '_', '.', '--' and '..' are forbidden, '-' and '.' cannot start or end the name, max length is 96:``

**Comment by CakeCrusher** [comment]
@piee-kun Hi sorry for the late reply. The loading bar unfortunately is broken and will never move, despite training actually making progress.

That is due to the utilization of `tqdm` in the `train.py` file in areas like the following: 
https://github.com/CakeCrusher/mimicbot/blob/039c95bd124cb788c46664b110ede937a572ce43/mimicbot_cli/train.py#L436-L437

The error is notifying you that after the first epoch, on the first save that the repo name is invalid. I am not sure if you explicitly set the repo name, if you did then try to change the name to the satisfy the requirements on the error, otherwise if you did not, I suggest explicitly setting a name (it is prompted at the beginning of the `forge` command). Please let me know if you did explicitly set the name (i may need to update the repo if you didn't).
https://github.com/d8ahazard/sd_dreambooth_extension/issues/626

Let me know how it goes.
